firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java

```diff
@@ -646,14 +633,49 @@ public void testEncodesMultipleFiltersOnDeeperCollections() {
   }
 
   @Test
-  public void testEncodesDisjunctiveFiltersOnDeeperCollections() {
+  public void testInSerialization() {
+    Query q =
+        Query.atPath(ResourcePath.fromString("rooms/1/messages/10/attachments"))
+            .filter(filter("tags", "in", asList("pending", "dimond")));
+    Target actual = serializer.encodeTarget(wrapQueryData(q));
+    ArrayValue.Builder inFilterValue =
+        ArrayValue.newBuilder()
+            .addValues(valueBuilder().setStringValue("pending"))
+            .addValues(valueBuilder().setStringValue("dimond"));
+
+    StructuredQuery.Builder structuredQueryBuilder =
+        StructuredQuery.newBuilder()
+            .addFrom(CollectionSelector.newBuilder().setCollectionId("attachments"))
+            .setWhere(
+                Filter.newBuilder()
+                    .setFieldFilter(
+                        FieldFilter.newBuilder()
+                            .setField(FieldReference.newBuilder().setFieldPath("tags"))
+                            .setOp(Operator.IN)
+                            .setValue(valueBuilder().setArrayValue(inFilterValue))))
+            .addOrderBy(defaultKeyOrder());
+    QueryTarget.Builder queryBuilder =
+        QueryTarget.newBuilder()
+            .setParent("projects/p/databases/d/documents/rooms/1/messages/10")
+            .setStructuredQuery(structuredQueryBuilder);
+    Target expected =
+        Target.newBuilder()
+            .setQuery(queryBuilder)
+            .setTargetId(1)
+            .setResumeToken(ByteString.EMPTY)
+            .build();
+
+    assertEquals(expected, actual);
+    assertEquals(serializer.decodeQueryTarget(serializer.encodeQueryTarget(q)), q);
+  }
```

Comment:
Any way we can make these tests less boilerplate-y?  Ideally it'd be much closer to the TypeScript one (https://github.com/firebase/firebase-js-sdk/blob/7a52f753c94d326e690ccafed48d82c8959dcf72/packages/firestore/test/unit/remote/node/serializer.test.ts#L786)

Basically, we should try to refactor so that we can test filter serialization in isolation (without a query, target, etc.).  Perhaps we should make `encodeRelationFilter()` public (or package-private?) so that we can test it directly?  Failing that perhaps we could have a test helper method that takes a filter and wraps it up in a dummy query / target in order to serialize / deserialize it.